metadata/src/utils/retain.rs

```diff
@@ -8,8 +8,9 @@ use crate::{
     ExtrinsicMetadata, Metadata, OuterEnumsMetadata, PalletMetadataInner, RuntimeApiMetadataInner,
     StorageEntryType,
 };
+use alloc::collections::BTreeMap;
+use hashbrown::HashSet;
```

Comment:
In other places I've just moved to using BTreeMap/BTreeSet to make it no-std compatible but hashbrown works too :)

Reply:
I specifically wanted to use a HashMap in this case because we do a lot of lookups and insertions on a fairly large set of type ids here. :)